front/src/services/helpers/formatSearchParamToPrismaQuery.ts

```diff
@@ -1,41 +1,50 @@
-import { Order, TimeOption } from "@/types/logDisplayOptions";
+import { LogDisplayOptions, Order, SortOptions, TimeOption } from "@/types/logDisplayOptions";
 import { llm_logs } from "@prisma/client";
 import {
     PrismaSort,
     SearchCondition,
     Timeframe,
 } from "../types/queryConditions";
+import { MS_PER_DAY, MS_PER_HOUR, MS_PER_WEEK } from "./timeConstants";
 
-const MS_PER_HOUR = 60 * 60 * 1000;
-const MS_PER_DAY = 24 * MS_PER_HOUR;
-const MS_PER_WEEK = 7 * MS_PER_DAY;
+export const convertSearchParamToPrismaConditions = (searchParams?: URLSearchParams) => {
+    const data = extractDataFromSearchParams(searchParams);
 
-export const extractDataFromSearchParam = (searchParams?: URLSearchParams) => {
-    const params = new URLSearchParams(searchParams);
-
-    const selectedTime =
-        (params.get("dateTimeFilter") as TimeOption) || "Last hour";
-    const timeframe = stringToTimeframeObject(selectedTime);
+    let timeframe: Timeframe;
+    if (data.dateTimeFilter === "Custom interval" && data.startDateTime && data.endDateTime)
+        timeframe = intervalToTimeframeObject(data.startDateTime, data.endDateTime);
+    else timeframe = stringToTimeframeObject(data.dateTimeFilter);
 
-    const sortBy = params.get("sortBy") || "datetime_utc";
-    const sortOrder = params.get("sortOrder") || "desc";
     const sortObject = stringsToSortObject(
-        sortBy as keyof llm_logs,
-        sortOrder as Order,
+        data.sortBy || "datetime_utc",
+        data.sortOrder || "desc",
     );
 
-    const searchString = params.get("search") || undefined;
-    const searchCondition = stringToSearchCondition(searchString);
+    const searchCondition = stringToSearchCondition(data.search);
 
     return { timeframe, sort: sortObject, searchCondition };
 };
 
+const extractDataFromSearchParams = (searchParams?: URLSearchParams): LogDisplayOptions => {
+    const params = new URLSearchParams(searchParams);
+    const data: LogDisplayOptions = {
+        dateTimeFilter: (params.get("dateTimeFilter") as TimeOption),
```

Comment:
(refactor) I didn't catch this sooner, but type casting with `as` should not be done in runtime

can we have some type guards?

Reply:
The `params.get()` method returns a `string | undefined` object, whatever the input is, so it has to be cast if we want to handle it correctly. This behaviour makes sense as the user can modify the url manually. Discussion to be had on slack / huddle / irl

Reply:
Conclusion of the huddle: for each types a safe cast should be implemented.